fvm/derived/derived_block_data.go

```diff
@@ -151,6 +151,13 @@ func (block *DerivedBlockData) GetProgramForTestingOnly(
 	return block.programs.GetForTestingOnly(addressLocation)
 }
 
+// CachedPrograms returns the number of programs cached.
+// Note: this should only be called after calling commit, otherwise
+// the count will contain invalidated entries.
+func (block *DerivedBlockData) CachedPrograms() int {
```

Comment:
Let me add that in another PR.

I think we want to keep track of that for scripts and transactions separately, and I'm also not clear if its best to keep track of that in programs directly, or offload it to the meter.

Reply:
fair enough.  In general, i think it may be better to count hit / misses on a per transaction level.

script programs aren't committed back into derived block data, so these won't be accounted for in the block total either.